tests/tracking/test_rest_tracking.py

```diff
@@ -751,7 +750,7 @@ def test_artifacts(mlflow_client, tmp_path):
     created_run = mlflow_client.create_run(experiment_id)
     assert created_run.info.artifact_uri.startswith(experiment_info.artifact_location)
     run_id = created_run.info.run_id
-    src_dir = tempfile.mkdtemp("test_artifacts_src")
+    src_dir = tmp_path.joinpath("test_artifacts_src")
```

Comment:
```suggestion
    src_dir = tmp_path.joinpath("test_artifacts_src").mkdir(parents=True, exist_ok=True)
```
Looks like line 754-756 can be changed to:

```
src_file = src_dir.joinpath("my.file")
src_file.write_text("Hello, World!")
```

to make the tests pass

Reply:
We can, but that's out of the scope of this PR :)

Reply:
AH that's right.... I totally forgot that os.path.join() can accept a pathlib.Path. That was why I suggested that change ;) It's not needed.

Reply:
wait, did this test fail?

Reply:
it did

Reply:
`mkdir` was missing, thanks for the catch!